packages/inference/test/vcr.ts

```diff
@@ -0,0 +1,163 @@
+import { isBackend, isFrontend } from "../src/utils/env-predicates";
+
+const TAPES_FILE = "./tapes.json";
+
+let VCR_MODE;
+
+/**
+ * Allows to record tapes with a token to avoid rate limit.
+ *
+ * If VCR_MODE is not set and a token is present then disable it.
+ */
+if (process.env.VCR_MODE) {
+	if ((process.env.VCR_MODE === "record" || process.env.VCR_MODE === "cache") && isFrontend) {
+		throw new Error("VCR_MODE=record is not supported in the browser");
+	}
+
+	VCR_MODE = process.env.VCR_MODE;
+} else {
+	VCR_MODE = process.env.HF_ACCESS_TOKEN ? "disabled" : "playback";
+}
+
+if (isFrontend) {
+	const originalFetch = window.fetch;
+
+	window.fetch = (...args) => {
+		return vcr(originalFetch, args[0], args[1]);
+	};
+} else {
+	const originalFetch = global.fetch;
+
+	global.fetch = (...args) => {
+		return vcr(originalFetch, args[0], args[1]);
+	};
+}
```

Comment:
I think instead of `window` and `global` you can use `globalThis` in both cases?